### ISSUE TITLE
Transition from D1 operator overload to D2 operator overloads

### ISSUE THREAD

**Issue by Geod24** [sociomantic-tsunami/ocean#747]
This needs to be done in v6.0.0, as D1 operators are virtual, while D2 are template instantiation (final), but can be made virtual if needed.
Some part of Ocean make use of it, e.g. [this](https://github.com/sociomantic-tsunami/ocean/blob/f253c5374d85948937fc8861804071b23752dacb/src/ocean/core/Enum.d#L153-L218).

**Comment by joseph-wakeling-frequenz** [comment]
Note that there are a few finnicky isues that come out of how various ocean functionality implements the older D1 operator overloads.

A quite common pattern is to have the implementation inside a local method, then use an alias of that to define the D1 operator overload.  The local implementations are in at least some cases overridden in subclasses (the only explicit example I found was in unittests, but we can't exclude that there are some other downstream examples).

See https://github.com/sociomantic-tsunami/ocean/pull/748 for an example of what I thought was the best way to deal with this -- I'm interested in yours (and others') feedback on this.

There are also some cases, like https://github.com/sociomantic-tsunami/ocean/pull/749, where the D1 operator definitions just seem _weird_, and make it look like they were added without really thinking through the use-cases.
